src/ui/FieldValue/FieldValue.ts

```diff
@@ -228,31 +238,34 @@ export class FieldValue extends Component {
     this.result = this.result || this.resolveResult();
     Assert.exists(this.result);
 
-    let loadedValueFromComponent = this.getValue();
-    if (loadedValueFromComponent == null) {
-      // Completely remove the element to ease stuff such as adding separators in CSS
+    if (TemplateFieldsEvaluator.evaluateFieldsToMatch(this.options.conditions, this.result) && this.getValue()) {
+      this.initialize();
+    } else {
       if (this.element.parentElement != null) {
         this.element.parentElement.removeChild(this.element);
       }
-    } else {
-      let values: string[];
-
-      if (isArray(loadedValueFromComponent)) {
-        values = loadedValueFromComponent;
-      } else if (this.options.splitValues) {
-        if (isString(loadedValueFromComponent)) {
-          values = map(loadedValueFromComponent.split(this.options.separator), (v: string) => {
-            return v.trim();
-          });
-        }
-      } else {
-        loadedValueFromComponent = loadedValueFromComponent.toString();
-        values = [loadedValueFromComponent];
-      }
-      this.appendValuesToDom(values);
-      if (this.options.textCaption != null) {
-        this.prependTextCaptionToDom();
+    }
+  }
+
+  private initialize() {
```

Comment:
Could be shortened to this:
```ts
private initialize() {
  this.appendValuesToDom(
    isArray(loadedValueFromComponent)
      ? <string[]>loadedValueFromComponent
      : (
        this.options.splitValues 
          ? (
            isString(loadedValueFromComponent)
              ? loadedValueFromComponent.split(
                this.options.separator
              ).map(value => value.trim())
              : undefined
          )
          : [
            <string>loadedValueFromComponent.toString()
          ]
      )
  );
  this.options.textCaption != null && this.prependTextCaptionToDom();
}
```

Reply:
I didn't want to modify that part, it's exactly as it was. If you want to refactor, feel free :).